### PR TITLE
[supervisor] Use a fallback default git credentials for cloning dotfiles

### PR DESCRIPTION
**Pull request by axonasif**
## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/16077 https://github.com/gitpod-io/gitpod/issues/14984

## Release Notes

```release-note
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@axonasif: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/supervisor/pkg/supervisor/supervisor.go [hunk 1]
```diff
@@ -544,11 +544,14 @@ func installDotfiles(ctx context.Context, cfg *Config, tokenService *InMemoryTok
 					Host: repoUrl.Host,
 					Kind: KindGit,
 				})
-				if err != nil {
-					return
+				if err == nil {
+					username = resp.User
+					password = resp.Token
+				} else {
+					username = "gitpod"
```
Comment (akosyakov) [inline components/supervisor/pkg/supervisor/supervisor.go:551]: i don't see the point. credential helper is using tokenService as well

Comment (axonasif) [reply components/supervisor/pkg/supervisor/supervisor.go:551]: Does that mean `git` would still use `gp credential-helper` while performing the clone?

**Pull request closed without merging**